www/js/onboarding/SaveQrPage.tsx

```diff
@@ -23,12 +24,22 @@ const SaveQrPage = ({  }) => {
   useEffect(() => {
     if (overallStatus == true && !registerUserDone) {
       logDebug('permissions done, going to log in');
-      login(onboardingState.opcode).then((response) => {
-        logDebug('login done, refreshing onboarding state');
-        setRegisterUserDone(true);
-        preloadDemoSurveyResponse();
-        refreshOnboardingState();
-      });
+      markConsented()
+        .then(login(onboardingState.opcode)
+          .then((response) => {
+            logDebug('login done, refreshing onboarding state');
+            setRegisterUserDone(true);
+            preloadDemoSurveyResponse();
+            refreshOnboardingState();
+
+            //fully consented, so can handle other aspects
+            //other plugins - previously used $emit
+            const PushNotify = getAngularService("PushNotify");
+            PushNotify.afterConsent();
+            const StoreSeviceSettings = getAngularService("StoreDeviceSettings");
+            StoreSeviceSettings.afterConsent();
```

Comment:
You can see my other comment below, but I think this is not the right place from which to make these calls

Reply:
I moved these calls into `markConsented` in `starPrefs`